plugins/woocommerce-blocks/assets/js/blocks/product-template/edit.tsx

```diff
@@ -281,20 +285,14 @@ const ProductTemplateEdit = (
 		[ products ]
 	);
 
-	const hasLayoutFlex = layoutType === 'flex' && columns > 1;
-	let customClassName = '';
-	if ( hasLayoutFlex ) {
-		const dynamicGrid = `wc-block-product-template__responsive columns-${ columns }`;
-		const staticGrid = `is-flex-container columns-${ columns }`;
-
-		customClassName = shrinkColumns ? dynamicGrid : staticGrid;
-	}
+	useEffect( () => {
+		setAttributes( { layout: templateLayout } );
+	}, [ setAttributes, templateLayout ] );
```

Comment:
Just to confirm, is this done to ensure that the layout attribute value of the Product Template block stays in sync with the Product Collection? 🙂

Reply:
Yeah :smile:

One caveat to using `supports.layout` is that the `layout` attribute needs to be configured _on the block it applies to_. Editing `layout` on the Product Collection block would impact the filters inside but not the product template gutter. The Query Loop block addressed this by moving the list/grid selection to the Post Template block.

Since we want to keep it on the Product Collection block, this mirrors the `templateLayout` to the Product Template block's `layout` so that Gutenberg can handle it for us correctly.